From aa0989a14fd4f5cec8c3ed2b48b25a0e5bf52fe6 Mon Sep 17 00:00:00 2001
From: Martin Kosek <mkosek@redhat.com>
Date: Thu, 14 Mar 2013 14:33:56 +0100
Subject: [PATCH 81/98] Use temporary CCACHE in ipa-client-install

ipa-client-install failed if user had set his own KRB5CCNAME in his
environment. Use a temporary CCACHE for the installer to avoid these
kind of errors.

https://fedorahosted.org/freeipa/ticket/3512
---
 ipa-client/ipa-install/ipa-client-install | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/ipa-client/ipa-install/ipa-client-install b/ipa-client/ipa-install/ipa-client-install
index 62149aa65716d2feb6bd5dd5071907f4a6918636..e198d8dcb335d9d4ee91c70fb793b7026dba4004 100755
--- a/ipa-client/ipa-install/ipa-client-install
+++ b/ipa-client/ipa-install/ipa-client-install
@@ -1911,6 +1911,9 @@ def install(options, env, fstore, statestore):
                 root_logger.error("Test kerberos configuration failed")
                 return CLIENT_INSTALL_ERROR
             env['KRB5_CONFIG'] = krb_name
+            (ccache_fd, ccache_name) = tempfile.mkstemp()
+            os.close(ccache_fd)
+            env['KRB5CCNAME'] = os.environ['KRB5CCNAME'] = ccache_name
             join_args = ["/usr/sbin/ipa-join", "-s", cli_server[0], "-b", str(realm_to_suffix(cli_realm))]
             if options.debug:
                 join_args.append("-d")
@@ -2024,6 +2027,10 @@ def install(options, env, fstore, statestore):
             except OSError:
                 root_logger.error("Could not remove %s", krb_name)
             try:
+                os.remove(ccache_name)
+            except OSError:
+                pass
+            try:
                 os.remove(krb_name + ".ipabkp")
             except OSError:
                 root_logger.error("Could not remove %s.ipabkp", krb_name)
-- 
1.8.3.1

